web-common/src/features/dashboards/dashboard-stores.ts

```diff
@@ -177,6 +180,26 @@ const metricViewReducers = {
         metricsExplorer.visibleDimensionKeys = new Set(
           metricsView.dimensions.map((dim) => dim.name)
         );
+
+        const dimensionsMap = getMapFromArray(
```

Comment:
I'm not following why we need a map. Can't we just use the array? Can you add a comment if this is an optimization?